libs/model-assessment/src/lib/ModelAssessmentDashboard/Cohort/GenerateTimeSeriesCohorts.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import {
+  ErrorCohort,
+  JointDataset,
+  IFilter,
+  FilterMethods,
+  Cohort,
+  CohortSource,
+  IDataset
+} from "@responsible-ai/core-ui";
+
+import { translatePreBuiltCohortFilterForDataset } from "./ProcessPreBuiltCohort";
+
+export function generateTimeSeriesCohorts(
+  defaultErrorCohort: ErrorCohort,
+  preBuiltErrorCohortList: ErrorCohort[],
+  dataset: IDataset,
+  jointDataset: JointDataset
+): ErrorCohort[] {
+  if (
+    dataset.feature_metadata?.time_series_id_features &&
+    dataset.feature_metadata?.time_series_id_features?.length > 0
+  ) {
+    // Need to generate time series as cohorts for forecasting.
+    const timeSeriesIdFeatureIndices =
+      dataset.feature_metadata?.time_series_id_features.map((featureName) =>
+        dataset.feature_names.indexOf(featureName)
+      );
+    const distinctTimeSeriesIds = new Set<string>();
+    dataset.features.forEach((row) => {
+      const timeSeriesId = timeSeriesIdFeatureIndices.map(
+        (index) => row[index]
+      );
+      distinctTimeSeriesIds.add(JSON.stringify(timeSeriesId));
+    });
+
+    preBuiltErrorCohortList = [];
+    distinctTimeSeriesIds.forEach((timeSeriesId) => {
+      const timeSeriesIdArray = JSON.parse(timeSeriesId);
+      const filterList: IFilter[] = [];
+      const cohortNameParts: string[] = [];
+      dataset.feature_metadata?.time_series_id_features?.forEach(
```

Comment:
remove the "?" ?

Reply:
The compiler expects the ? operator in this line (both occurrences). I suspect it's because we're inside the forEach (?)